conductor_api/src/holochain.rs

```diff
@@ -168,26 +168,51 @@ impl Holochain {
         let mut instance = Instance::from_state(loaded_state.clone(), context.clone());
         let new_context = instance.initialize(None, context.clone())?;
         Ok(Holochain {
-            instance,
-            context: new_context.clone(),
+            instance: Some(instance),
+            context: Some(new_context.clone()),
             active: false,
         })
     }
 
+    fn check_instance(&self) -> Result<(), HolochainInstanceError> {
+        if self.instance.is_none() || self.context.is_none() {
+            Err(HolochainInstanceError::InstanceNotActiveYet)
+        } else {
+            Ok(())
+        }
+    }
+
+    fn check_active(&self) -> Result<(), HolochainInstanceError> {
+        if !self.active {
+            Err(HolochainInstanceError::InstanceNotActiveYet)
+        } else {
+            Ok(())
+        }
+    }
```

Comment:
Now we have two ways to return `InstanceNotActiveYet`, one if the `active` bool is false, another if the instance or context is not initialized yet. They mean different things, because an instance can be initialized and yet not active.

However, I don't think we actually use the notion of "active" for anything meaningful yet. What would it mean for an instance to be initialized and not active? Currently the only thing that changes is this boolean. It feels like a placeholder for something more meaningful. If we are going to add a more meaningful notion of active/inactive, maybe we should do it now, or soon as a followup. I imagine deactivating an instance also has to do with stopping its action loop, without totally tearing it down, which is related to this work.

Reply:
Yes. „active“ should reflect if the action loop is running. I was reusing InstanceNotActiveYet but intended to add a new error.